src/apps/activity_assignments/errors.py

```diff
@@ -4,4 +4,16 @@
 
 
 class ActivityAssignmentActivityOrFlowError(ValidationError):
+    message = _("Either activity_id or activity_flow_id must be provided, but not both")
+
+
+class ActivityAssignmentNotActivityAndFlowError(ValidationError):
```

Comment:
The reason two distinct errors weren't needed by `ActivityAssignmentCreate` previously was because @rcmerlo was just overriding the error message in the [second error case](https://github.com/ChildMindInstitute/mindlogger-backend-refactor/pull/1556/files#diff-36996b2475a09bd40766f2ca5e5586ac3b6f0ab90431fc82a795ef8f6145f9dcR27). It's fine to change it to what you've done here (two distinct error classes), but just make sure to also update `ActivityAssignmentCreate` to follow suit for consistency.

Reply:
+1